source/blog/2017-08-04-ovirt-webadmin-gwt-debug-quick-refresh.html.md

```diff
@@ -0,0 +1,132 @@
+---
+title: oVirt webadmin gwt debug quick refresh
+author: gshereme
+tags: GWT, webadmin, debugging
+date: 2017-08-04 15:00:00 CET
+comments: true
+published: true
+---
+
+As a developer, one drawback of using GWT for the oVirt webadmin is that the GWT compile process takes an exceptionally long time. If you make a change in some code and rebuild the ovirt-engine project using make, you'll be waiting several minutes to test your change. In practice, this long of a delay in the usual code-compile-refresh-test cycle would be unbearable.
+
+READMORE
+
+Luckily, we can use GWT Super Dev Mode ("SDM") to start up a quick-refresh capable instance of the application. With SDM running, you can make a change in GWT and test the refreshed change within seconds.
+
+The below approach will only work if you're intested in quick refreshes only. If you want to step through code, oVirt and SDM don't work well together because of oVirt's code and source map size. Therefore, we'll disable source maps below. (You can still debug and step through code using GWT Classic Dev Mode, which isn't covered in this post.)
```

Comment:
also, re: (You can still debug and step through code using GWT Classic Dev Mode, which isn't covered in this post.)

that's an entire parenthetical sentence, and the period goes inside the parentheses in that case :D